Assets/Scripts/ViconNexusUnityStream/DataStreamer.cs

```diff
@@ -1,56 +1,78 @@
-using System;
-using System.Collections;
 using System.Collections.Generic;
-using System.Linq;
 using NativeWebSocket;
 using ubco.ovilab.ViconUnityStream;
 using UnityEngine;
 using System.Text;
 using Newtonsoft.Json;
-using MessagePack;
-using MessagePack.Resolvers;
 using Newtonsoft.Json.Linq;
-using UnityEngine.Serialization;
 
 public class DataStreamer : Singleton<DataStreamer>
 {
     [SerializeField] private string baseURI = "ws://viconmx.hcilab.ok.ubc.ca:5001/";
-    public List<string> subjectList;
-    public WebSocket signalStream => webSocket;
     public Dictionary<string, Data> StreamedData => data;
     public Dictionary<string, string> StreamedRawData => rawData;
     
+    private List<string> subjectList;
     private WebSocket webSocket;
     private Dictionary<string, Data> data = new();
     private Dictionary<string, string> rawData = new();
-    private async void Start()
+
+    /// <inheritdoc />
+    private void OnEnable()
+    {
+        SetupConnection();
+    }
+
```

Comment:
Maybe move this into Awake() or Start() so this function is called only after the Custom Subject Scripts register their respective subjects.

Reply:
That's a good point. OnEnable should check the registered subjects before setting things up.